### ISSUE TITLE
Persistent style objects

### ISSUE THREAD

**Issue by rtsao** [hellofresh/css-in-js-perf-tests#31]
Depending on the use case, style objects may or may not be defined outside of render. With the current benchmark suite, they are. This means a WeakMap cache can essentially "defeat" the benchmarks, if the purpose of the benchmark is to determine the performance for initial render.

@threepointone had a good write up here: https://github.com/threepointone/glamor/blob/master/docs/weakmaps.md

So I think the benchmarks should be refactored to handle both cases, one where reference equality of style objects is maintained between render, and another where it is not. So we sort of benchmark "initial render" and "subsequent render". We probably want a warm-up run for the subsequent render case so any caching will already exist.

**Comment by threepointone** [comment]
To be fair, this benchmark doesn't actually measure initial render for dynamic styles either. And doesn't take stuff like SSR/rehydrating into account (despite running in a server, ha). And more :) I feel like this benchmark measures just one tiny bit of all the things that could contribute to perf. If you're planning on a refactor, do consider being more 'real world'. Plsthx!

**Comment by rtsao** [comment]
@threepointone Yeah, client-side benchmarks are obviously needed and this has been discussed in the previous issue here: https://github.com/hellofresh/css-in-js-perf-tests/issues/18

**Comment by kof** [comment]
I think the point of this benchmark should be to help users decide which library to use. Currently I don't think they can do this based on this perf tests.

**Comment by kof** [comment]
Another point is probably help cssinjs authors compare their solutions and improve, but this isn't working either, we just try to outsmart the bench)
